go/oasis-node/cmd/debug/txsource/workload/commission.go

```diff
@@ -0,0 +1,251 @@
+	nextAlignedEpoch := currentEpoch + c.rules.RateBoundLead + 2
+	// Find first next epoch aligned with rateChangeInterval.
+	for ; nextAlignedEpoch%c.rules.RateChangeInterval != 0; nextAlignedEpoch++ {
+	}
+
+	maxBoundSteps := c.rules.MaxBoundSteps
+	maxRateSteps := c.rules.MaxRateSteps
+	// Check existing bounds/steps. In case there are existing steps with
+	// epoch < nextAlignedEpoch, those cannot get amended, therefore we need to
+	// count those to not go over maxAllowed limits.
+	for _, step := range commissionSchedule.Bounds {
+		if step.Start < nextAlignedEpoch {
+			maxBoundSteps--
+			continue
+		}
+		break
+	}
+	for _, step := range commissionSchedule.Rates {
+		if step.Start < nextAlignedEpoch {
+			maxRateSteps--
+			continue
+		}
+		break
+	}
+
+	// Select bound step epochs.
+	var boundStepEpochs []epochtime.EpochTime
+	nBoundSteps := rng.Intn(maxBoundSteps-1) + 1
+	for i := 0; i < nBoundSteps; i++ {
+		boundStepEpochs = append(boundStepEpochs, nextAlignedEpoch+(epochtime.EpochTime(i)*c.rules.RateChangeInterval))
+	}
+	// Select rate step epochs.
+	var rateStepEpochs []epochtime.EpochTime
+	nRateSteps := rng.Intn(maxRateSteps-1) + 1
+	for i := 0; i < nRateSteps; i++ {
+		rateStepEpochs = append(rateStepEpochs, nextAlignedEpoch+(epochtime.EpochTime(i)*c.rules.RateChangeInterval))
+	}
+
+	var amendSchedule staking.AmendCommissionSchedule
+	// Generate bounds.
+	for i, boundEpoch := range boundStepEpochs {
+		// In case we are setting more bound steps than rates, make sure that
+		// bounds after last rate step are not exclusive, as the last rate rule
+		// will need to be within all following bounds.
+		var maxBound, minBound int64
+		if i >= nRateSteps {
+			prevMaxBound := amendSchedule.Amendment.Bounds[i-1].RateMax.ToBigInt().Int64()
+			prevMinBound := amendSchedule.Amendment.Bounds[i-1].RateMin.ToBigInt().Int64()
+			// [prevMinBound, prevMaxBound]
+			maxBound = rng.Int63n(prevMaxBound-prevMinBound+1) + prevMinBound
+			// [prevMinBound, newMaxBound]
+			minBound = rng.Int63n(maxBound-prevMinBound+1) + prevMinBound
+		} else {
+			// [10, 100_000]
+			maxBound = rng.Int63n(100_000-10+1) + 10
+			// [0, maxBound]
+			minBound = rng.Int63n(maxBound + 1)
+		}
+		bound := staking.CommissionRateBoundStep{
+			Start: boundEpoch,
+		}
+		if err = bound.RateMin.FromInt64(minBound); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+		if err = bound.RateMax.FromInt64(maxBound); err != nil {
+			return fmt.Errorf("Rate.FromInt64 err: %w", err)
+		}
+		amendSchedule.Amendment.Bounds = append(amendSchedule.Amendment.Bounds, bound)
+	}
+
+	// Genarete rates.
+	for i, rateEpoch := range rateStepEpochs {
+		// Get bounds for i'th epoch.
+		var bound staking.CommissionRateBoundStep
+		if i < nBoundSteps {
+			bound = amendSchedule.Amendment.Bounds[i]
+		} else {
+			// In case there are more rate steps than bounds, last bound will
+			// be in effect.
+			bound = amendSchedule.Amendment.Bounds[nBoundSteps-1]
```

Comment:
is it impossible for the bounds amendment to be zero-length?

Reply:
outdated / addressed by changing the rate generation